app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt

```diff
@@ -62,17 +73,60 @@ class CalculatorPresenter(
             } else if (!model.first_operand.contains(DECIMAL_POINT)) {
                 model.first_operand = "${model.first_operand}$DECIMAL_POINT"
             } else {
-                view.showDecimalError()
+                view.showMessage(CalculatorError.DECIMAL_ERROR)
             }
         } else if (model.operator != EMPTY_CHAR) {
             if (model.second_operand.isEmpty()) {
                 model.second_operand = "$NUMBER_ZERO$DECIMAL_POINT"
             } else if (!model.second_operand.contains(DECIMAL_POINT)) {
                 model.second_operand = "${model.second_operand}$DECIMAL_POINT"
             } else {
-                view.showDecimalError()
+                view.showMessage(CalculatorError.DECIMAL_ERROR)
             }
         }
         updateVisor()
     }
+
+    private fun calculate(): String {
+        when (model.operator) {
+            OPERATOR_PLUS -> {
+                return decimalFormat.format(model.first_operand.toDouble() + model.second_operand.toDouble())
+            }
+            OPERATOR_SUBSTRACTION -> {
+                return decimalFormat.format(model.first_operand.toDouble() - model.second_operand.toDouble())
+            }
+            OPERATOR_MULTIPLY -> {
+                return decimalFormat.format(model.first_operand.toDouble() * model.second_operand.toDouble())
+            }
+            OPERATOR_DIVIDE -> {
+                if (!model.second_operand.equals(NUMBER_ZERO)) {
+                    return decimalFormat.format(model.first_operand.toDouble() / model.second_operand.toDouble())
+                } else {
+                    view.showMessage(CalculatorError.MATH_ERROR)
+                    onClearButtonPressed()
+                }
+            }
+        }
+        return EMPTY_STRING
+    }
+
+    override fun onEqualPressed() {
+        if (!((model.first_operand.isEmpty()) || (model.second_operand.isEmpty()) || (model.operator == EMPTY_CHAR))) {
```

Comment:
you can use here 
((model.first_operand.isNotEmpty()) && (model.second_operand.isNotEmpty())
instad of
!((model.first_operand.isEmpty()) || (model.second_operand.isEmpty())

if I'm not wrong it gives more performance

Reply:
Yes, you are right. Thanks Mau, done!